server/memphis_handler_schemas.go

```diff
@@ -0,0 +1,368 @@
+	}
+	newSchemaVersion := models.SchemaVersion{
+		ID:            primitive.NewObjectID(),
+		VersionNumber: versionNumber,
+		Active:        active,
+		CreatedByUser: user.Username,
+		CreationDate:  time.Now(),
+		SchemaContent: schemaContent,
+	}
+	newSchema := models.Schema{
+		ID:        primitive.NewObjectID(),
+		Name:      schemaName,
+		Type:      schemaType,
+		Versions:  schema.Versions,
+		IsDeleted: false,
+	}
+
+	filter = bson.M{"name": newSchema.Name, "is_deleted": false}
+	update := bson.M{
+		"$setOnInsert": bson.M{
+			"_id":  newSchema.ID,
+			"name": newSchema.Name,
+			"type": newSchema.Type,
+		},
+		"$push": bson.M{"versions": newSchemaVersion.ID},
+	}
+	opts := options.Update().SetUpsert(true)
+	updateResults, err := schemasCollection.UpdateOne(context.TODO(), filter, update, opts)
+	if err != nil {
+		serv.Errorf("CreateSchema error: " + err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		return
+	}
+	if updateResults.MatchedCount > 0 {
+		serv.Warnf("Scheam with the same name is already exist")
+		newSchema.ID = schema.ID
+	} else {
+		message := "Schema " + schemaName + " has been created"
+		serv.Noticef(message)
+		var auditLogs []interface{}
+		newAuditLog := models.AuditLog{
+			ID:            primitive.NewObjectID(),
+			StationName:   schemaName,
+			Message:       message,
+			CreatedByUser: user.Username,
+			CreationDate:  time.Now(),
+			UserType:      user.UserType,
+		}
+		auditLogs = append(auditLogs, newAuditLog)
+		err = CreateAuditLogs(auditLogs)
+		if err != nil {
+			serv.Warnf("CreateSchema error: " + err.Error())
+		}
+	}
+
+	filter = bson.M{"_id": newSchemaVersion.ID, "is_deleted": false}
+	update = bson.M{
+		"$setOnInsert": bson.M{
+			"_id":             newSchemaVersion.ID,
+			"version_number":  newSchemaVersion.VersionNumber,
+			"active":          newSchemaVersion.Active,
+			"created_by_user": newSchemaVersion.CreatedByUser,
+			"creation_date":   newSchemaVersion.CreationDate,
+			"schema_content":  newSchemaVersion.SchemaContent,
+		},
+	}
+	_, err = schemasVersionCollection.UpdateOne(context.TODO(), filter, update, opts)
+
+	if err != nil {
+		serv.Errorf("CreateSchema error: " + err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		return
+	}
+	newSchema.Versions = append(newSchema.Versions, newSchemaVersion.ID)
+	c.IndentedJSON(200, newSchema)
+}
+
+func (sh SchemasHandler) GetAllSchemas(c *gin.Context) {
+	schemas, err := sh.GetAllSchemasDetails()
+	if err != nil {
+		serv.Errorf("GetAllSchemas error: " + err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		return
+	}
+	c.IndentedJSON(200, schemas)
+}
+
+func (sh SchemasHandler) GetSchemaDetails(c *gin.Context) {
+	var body models.GetSchemaDetails
+	ok := utils.Validate(c, &body, false, nil)
+	if !ok {
+		return
+	}
+	schemaName := strings.ToLower(body.SchemaName)
+	schemaDetails, err := sh.GetSchemaDetailsBySchemaName(schemaName, body.VersionNumber)
```

Comment:
as we said version number is not needed cause this endpoint should return all versions